### PR TITLE
Back pressure fix from 3.7

### PR DESCRIPTION
**Pull request by bentoi**
This PR merges the back pressure fix from 3.7 to main.

### CODE REVIEW

File: csharp/test/Ice/ami/AllTests.cs [hunk 1]
```diff
@@ -929,6 +929,37 @@ public static async Task allTestsAsync(global::Test.TestHelper helper, bool coll
...
 
+                output.Write("testing back pressure... ");
+                output.Flush();
+                {
+                    // Keep the 3 server thread pool threads busy.
+                    Task sleep1Task = p.sleepAsync(1000);
+                    Task sleep2Task = p.sleepAsync(1000);
+                    Task sleep3Task = p.sleepAsync(1000);
+                    bool canceled = false;
+                    using(var cts = new CancellationTokenSource(200))
```
Comment (bernardnormier) [inline csharp/test/Ice/ami/AllTests.cs:940]: Use the new using statement syntax.

File: csharp/test/Ice/ami/AllTests.cs [hunk 2]
```diff
@@ -929,6 +929,37 @@ public static async Task allTestsAsync(global::Test.TestHelper helper, bool coll
...
+                    {
+                        try
+                        {
+                            var onewayProxy = (Test.TestIntfPrx)p.ice_oneway();
+
+                            // Sending should be canceled because the TCP send/receive buffer size on the server is set
+                            // to 50KB. Note: we don't use the cancel parameter of the operation here because the
+                            // cancellation doesn't cancel the operation whose payload is being sent.
+                            onewayProxy.opWithPayloadAsync(new byte[768 * 1024]).Wait(cts.Token);
```
Comment (bernardnormier) [inline csharp/test/Ice/ami/AllTests.cs:949]: Please use await

File: csharp/test/Ice/ami/AllTests.cs [hunk 3]
```diff
@@ -929,6 +929,37 @@ public static async Task allTestsAsync(global::Test.TestHelper helper, bool coll
...
+                        }
+                        catch(OperationCanceledException)
```
Comment (bernardnormier) [inline csharp/test/Ice/ami/AllTests.cs:951]: The formatting looks incorrect.

File: csharp/test/Ice/ami/AllTests.cs [hunk 4]
```diff
@@ -929,6 +929,37 @@ public static async Task allTestsAsync(global::Test.TestHelper helper, bool coll
...
+                        {
+                            canceled = true;
+                        }
+                    }
+                    test(canceled && !sleep1Task.IsCompleted);
+                    sleep1Task.Wait();
```
Comment (bernardnormier) [inline csharp/test/Ice/ami/AllTests.cs:957]: Should be all await.

**Review by bernardnormier — COMMENTED** [review]
I don't understand how the idleTimeout tests works.

Since back pressure works as of this PR, and the idle check is currently back-pressure unaware, the first test should fail.

**Comment by bernardnormier** [comment]
> I don't understand how the idleTimeout tests works.
> 
> Since back pressure works as of this PR, and the idle check is currently back-pressure unaware, the first test should fail.

The test fails, fortunately:
```
Ice.ConnectionLostException: Exception of type 'Ice.ConnectionLostException' was thrown.
   at Ice.Internal.OutgoingAsyncBase.invokeException() in /Users/runner/work/ice/ice/csharp/src/Ice/Internal/OutgoingAsync.cs:line 95
--- End of stack trace from previous location ---
   at Ice.idleTimeout.AllTests.testIdleCheckDoesNotAbortConnectionWhenThreadPoolIsExhausted(TestIntfPrx p, TextWriter output) in /Users/runner/work/ice/ice/csharp/test/Ice/idleTimeout/AllTests.cs:line 41
   at Ice.idleTimeout.AllTests.allTests(TestHelper helper) in /Users/runner/work/ice/ice/csharp/test/Ice/idleTimeout/AllTests.cs:line 17
   at Ice.idleTimeout.Client.runAsync(String[] args) in /Users/runner/work/ice/ice/csharp/test/Ice/idleTimeout/Client.cs:line 12
   at Test.TestDriver.runTestAsync[T](String[] args) in /Users/runner/work/ice/ice/csharp/test/TestCommon/TestHelper.cs:line 232

test in Ice/idleTimeout failed:
```

Please comment it out for now.

**Comment by bentoi** [comment]
Fixed the CI failures:
- server idle time bug
- FD limit bug
I've also disabled the failing idle timeout test.

**Review by bernardnormier — APPROVED** [review]

**Pull request merged**